TestingTool/actions.py

```diff
@@ -3,31 +3,48 @@
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.common.exceptions import TimeoutException
 
-
 # def example_action(action_args, browser_instance):
 #     some_arg = action_args.get("some_arg")
 #     if some_arg is None:
 #         return False, {"err": "Lipseste argumentul some_arg"}
 
 
-def click_element(action_args, browser_instance):
+def element_from_xpath(action_args, browser_instance):
     x_path_val = action_args.get("xpath")
     if x_path_val is None:
         return False, {"err": "You didnt give me an xpath in action_args"}
-
     wait = WebDriverWait(browser_instance, 2)
     try:
         element = wait.until(EC.presence_of_element_located(
             (By.XPATH, x_path_val)
         ))
     except TimeoutException:
-        return False, {"err": "timeout exception when looking for element at given xpath"}
+        return False, {"err": "Timeout exception when looking for element at given xpath"}
+    return True, element
+
+
+def click_element(action_args, browser_instance):
+    result, data = element_from_xpath(action_args, browser_instance)
+    if result is False:
+        return result, data
+    data.click()
+    return True, {"success": "Clicked on xpath given in action_args"}
+
 
-    element.click()
-    return True, {"success": "clicked on xpath given in action_args"}
+def input_element(action_args, browser_instance):
+    result, data = element_from_xpath(action_args, browser_instance)
+    input_text = action_args.get("input_text")
+    if result is False:
+        return result, data
+    if input_text is None:
+        return False, {"err": {"You didnt give me an input text in action_args"}}
```

Comment:
Aici obiectul returnat ca si "err" nu ar trebui sa fie un dict, ci un string

Reply:
fuck it, am apasat gresit